### DISCUSSION TITLE
Error in installing RAPIDS on Mac

### DISCUSSION THREAD

**Discussion by chetkar** [carissalow/rapids#179] [Q&A]
Dear All,

I am getting error while installing RAPIDS on Mac. In particular, I am getting error while executing following commands
1. snakemake -j1 renv_install
2.snakemake -j1 renv_restore

Please find the screenshot of the error output attached. In particular, the error message is stating "KeyError in line 4 of rules/models.smk: "PARAMS_FOR_ANALYSIS". Could you please provide help and guidance?

Looking forward to hearing from you,
Chetkar

![Error](https://user-images.githubusercontent.com/23512904/158436881-62425c69-ed1c-4e62-8e51-cdf3f0b46213.jpg)

**Comment by JulioV** [comment]
Hi, that is unexpected behavior and will try to reproduce it on my side. In the meantime feel free to try our Docker container as an alternative installation method

**Reply by JulioV** [reply]
@chetkar I could not replicate this problem in MacOS 12.2 on an M1.  I can suggest you try to install RAPIDS again from step 5 and double check you are running brew and installing R and miniconda using Rosetta.

**Reply by chetkar** [reply]
> Hi, that is unexpected behavior and will try to reproduce it on my side. In the meantime feel free to try our Docker container as an alternative installation method

Hi Julio,

Thank you for your quick response. I did try running Docker on Mac. I set up the files under rapids on the Docker container as in this folder (https://www.dropbox.com/sh/tvmyhjeqficwiy4/AAAGjmwwIElt2derCfyB9BSSa?dl=0). 

However, executing rapids (using ./rapids -j1) does run the rapids but it doesn't write any output. Could you please throw some light on what I am doing wrong?

Looking forward to hear from you.
Chetkar

**Reply by JulioV** [reply]
It’s difficult to troubleshoot the problem without more information (it sounds like your device ids might be incorrect in your participant files). You could:

- start again with the minimal example in the docs and add sensors one by one
- Temporarily disable sensors and participants until you get the expected results or are left with one of each that will help you diagnose the problem and open a GitHub issue.
- Open a GitHub issue and follow the template to include as much information as possible. We ask that you take advantage of GitHub’s code markdown syntax and avoid linking to external sites like Dropbox. This is so users in the future with the same problem can get the full context of the problem and its solution.

**Reply by chetkar** [reply]
Hey, I take your larger point on the Github's code markdown. I would appreciate though if you can clarify the following
1. Do we need different SnakeFiles for different datasets?

**Reply by JulioV** [reply]
No, the only files you need to modify if you are processing our stock features and sensors are

- the config file 
- participant files
- time segment file

You can find detailed instructions here: https://www.rapids.science/1.8/setup/configuration/

**Reply by JulioV** [reply]
Hi @chetkar, did you find the cause of this issue?
